### DISCUSSION TITLE
Shared folder not working with WinXP guest

### DISCUSSION THREAD

**Discussion by ianlewisau** [utmapp/UTM#3625] [UTM Help (macOS)]
Hi all,

I have installed UTM (latest) on 2 macbooks, one an Intel CPU and the other Apple M1. I've installed the same WinXP guest on both along with the Spice Guest Tools package. Everything works, seemingly with the exception of the shared folder. I can set the shared folder, but I don't see anything in the Computer view in the guest. This is the same on both hosts. (Why WinXP? It's because of a 16bit application that I still require.) The only way I've been able to transfer files is via USB, however this is a little tedious. My preference is to not enable network access for the WinXP guest as it's not required, and also due to the security issues with doing so. Does anyone have any thoughts as to what the issue might be?

**Comment by fastfading** [comment]
same issue , here ,  m1 chips  
any one fix this ?

**Reply by ianlewisau** [reply]
It seems that there aren't many using WinXP any longer. Fortunately, I don't have to use it often, so the USB workaround is acceptable - just a little annoying.

**Reply by cesaravila79** [reply]
Have you tried the tip on the following link?
https://github-wiki-see.page/m/utmapp/UTM/wiki/Useful-tips

**Reply by ianlewisau** [reply]
I tried to start my formerly working WinXP and it no longer starts. I haven't updated UTM, so I'm not sure what has happened. I'm guessing some security change in a MacOs update since the last time I ran it, but I can't see anything in logs that give any hint as to what's going on. Looks like I might have to set up from scratch again.

**Reply by nikshilov** [reply]
Finally launched Sims 1 on m1 air. Solved spice tools did not helped at all, so I created a CDR image from a folder using Disk Utility and then applied the command below to get the ISO image, which you can easily inject into virtual cd-rom.


`hdiutil makehybrid -iso -joliet -o -FILENAME.iso FILENAME.cdr`

**Reply by gingerbeardman** [reply]
Also having this problem on XP and also 98SE (for some old old software).

**Reply by nonno** [reply]
Same problem with the shared folder, I tried also to do the workaround with a usb stick but Windows XP didn't recognize it.

**Reply by gingerbeardman** [reply]
I ended up buying a little PC to run my old XP and 98 VMs. Easier than workarounds for UTM.

**Reply by AtP-365** [reply]
I had the same issue, with a "clean" install of WinXP. I was baffled, as I had used it a number of times. 

Then, quite by accident, what I found to work was to have mounted spice-guest-tools-0.164.4 as a CD within the UTM interface (at the bottom). And it loaded fine. Hope that may help someone.

**Comment by ianlewisau** [comment]
One way that works is to share a folder from XP and then attach to it from Finder on the Mac (connect to smb://your_utm_ip. It's not as convenient as the UTM folder sharing would be, but it works well enough.

**Reply by nonno** [reply]
Excuse me for the silly question, I'm a bit lost. I shared a folder on XP, switched off the firewall, got computer name and IP. From Mac side I tried "Connect to server" from Finder combining `smb://` and then IP, IP/folder_name, computer_name, computer_name/folder_name... nothing. I cannot find anything on the web, I don't know if it's my UTM machine that needs some adjustments on network settings (I got it from UTM website) or if I'm just doing some mistake somewhere else.

**Reply by gingerbeardman** [reply]
@nonno what is the IP address of your UTM VM, visible in emulated Windows XP?

**Reply by nonno** [reply]
`10.0.2.15`
<img width="528" alt="Screenshot 2022-09-07 at 15 52 34" src="https://user-images.githubusercontent.com/4105325/188895683-5647f5af-fc03-42be-9256-b43d8d743bdf.png">

**Reply by gingerbeardman** [reply]
OK, it will work if you change UTM settings: Network Mode = Shared Network 

Then your IP address will be 192.168.x.y and you will have success. 

![Screen shot 2022-09-07 at 15 41 08](https://user-images.githubusercontent.com/49612/188906865-d25bb3d9-7755-4448-b349-96f78523ac15.png)

I just did it this way, connecting directly to smb://192.168.x.y as Guest.

![Screen shot 2022-09-07 at 15 38 37](https://user-images.githubusercontent.com/49612/188906197-b0dbbec3-bef4-4661-a98b-8aaf6c4134a8.png)

It will also show up in the Network window on macOS:
![Screen shot 2022-09-07 at 15 39 04](https://user-images.githubusercontent.com/49612/188906310-f0b79e9a-409b-4b9c-9843-e19752328396.png)

**Reply by nonno** [reply]
It worked like charm, thank you so much @gingerbeardman. Sorry, I could have played with the settings but I was a bit worried to "break it" because at the beginning I also had issues while installing it 😅

**Reply by s4m13337** [reply]
I tried to connect this way but I am not able to connect as Guest. I tried using the windows xp's user account name and password, but that doesn't work either.

**Reply by nonno** [reply]
I don't connect as guest, but as Registered User, and on my virtual XP I only have the Administrator without password as user.
<img width="453" alt="Screenshot 2023-02-13 at 22 20 32" src="https://user-images.githubusercontent.com/4105325/218483579-1f2c4a74-177f-4bb1-90bc-045bce53e5b3.png">
<img width="508" alt="Screenshot 2023-02-13 at 22 21 25" src="https://user-images.githubusercontent.com/4105325/218483601-02a28158-504b-450e-88a8-b27045fda565.png">
<img width="450" alt="Screenshot 2023-02-13 at 22 21 40" src="https://user-images.githubusercontent.com/4105325/218483605-c5091ce9-3ea2-4749-9011-27e9d6b9371a.png">

**Reply by iamyeizi** [reply]
Name: Administrator
Password: "don't type anything" (if you don't registered a password)
Press Connect

**Reply by mihaitodor** [reply]
Works OK, but you'll need to run the Windows file sharing wizard to set up a shared folder and, also, disable the Windows Firewall. The main issue is that the transfer speeds between the host and guest are quite slow...

**Reply by 0rphee** [reply]
heeeelp! does anyone know what options do I need to set in the Windows network setup wizard? I've disabled the firewall, the workgroup is the same, and I've already shared a folder, but the host mac can't find anything at the ip address... 
EDIT: lol, forget it, I typed the ip on macos incorrectly


<img width="618" alt="Screenshot 2024-07-13 at 1 28 31 p m" src="https://github.com/user-attachments/assets/63db7696-d201-4d2d-825b-856405e400a2">

**Reply by DanilAtyukov-dev** [reply]
> heeeelp! does anyone know what options do I need to set in the Windows network setup wizard? I've disabled the firewall, the workgroup is the same, and I've already shared a folder, but the host mac can't find anything at the ip address... EDIT: lol, forget it, I typed the ip on macos incorrectly
> 
> <img alt="Screenshot 2024-07-13 at 1 28 31 p m" width="618" src="https://private-user-images.githubusercontent.com/79347623/348499690-63db7696-d201-4d2d-825b-856405e400a2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2MjkwNTcsIm5iZiI6MTcyMzYyODc1NywicGF0aCI6Ii83OTM0NzYyMy8zNDg0OTk2OTAtNjNkYjc2OTYtZDIwMS00ZDJkLTgyNWItODU2NDA1ZTQwMGEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDA5NDU1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0MjM3MzhlZDEwNTQzMmUzYWQ3NWE3ZTQ1Y2VlMTYxNzIwZWE3MTZlOTYyNDY2YTk1ZDYyMzkxMjNhZDczODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JDuePqpgWRCyzaASFvj5qnWeh2CBaIGpTxPt0jPRDio">

How'd you end up doing it? I can't do it either.

**Comment by dayue1984** [comment]
same here, seems that the windows xp does not support the weddav service

**Reply by gingerbeardman** [reply]
Apparently XP does support WebDAV https://www.ibm.com/docs/el/b2b-integrator/5.2?topic=client-configure-webdav-windows-xp

**Reply by nonno** [reply]
In my case the service was already running at startup.

**Comment by finchco** [comment]
FWIW still having the same problem with a Windows 7 guest.

**Reply by matt-telum** [reply]
Here's what I did and it works. I followed the instructions given by @gingerbeardman and was able to get the connection made from my host (m1 macbook) to guest (xp VM on UTM) but I couldn't get it to work the other way. 2 stupid things I forgot to enable... 

On Mac

Settings - Network - Advanced for the network you're connected to - WINS - set your workgroup to a name (you'll set the XP VM to the same one)
Settings - General - Sharing - File Sharing (turn on) - Options - Enable Windows file sharing (SMB). Set User/PW, done. 

On XP,

 it appears in My Network Places - Entire Network - Microsoft Windows Network - <Workgroup> - <Mac Name> - shares

hope this helps

**Comment by DougJoseph** [comment]
I had installed a fresh UTM VM guest instance of Windows XP Pro from an .iso, and installed the Spice Guest Tools. However, I had no connectivity at all between the host and the guest, neither clipboard, nor shared directory, or any network connections. I happened to stumble upon a mention in the guest OS that its settings were blocking all network access to and from. I was able to use a network setup 'wizard' in XP to make it so that a folder shared in the guest could be connected to on the host side via SMB. See pics. When I was asked to create a Workgroup name, I used the word "WORKGROUP." 

However, I still have no copy and paste from host to guest via clipboard, and my 'shared directory' option in the UTM settings is not working. 

I'm new to UTM and thankful for any advice. Can anyone advise me? 

<img width="1728" alt="Screenshot 2023-08-31 at 6 28 17 PM" src="https://github.com/utmapp/UTM/assets/8921397/dc03c3c6-a105-4924-9894-e4d50af54e7c">

<img width="914" alt="Screenshot 2023-08-31 at 6 28 24 PM" src="https://github.com/utmapp/UTM/assets/8921397/f13017f4-f96c-4bf2-89d8-759a4e5a86e0">

<img width="688" alt="Screenshot 2023-08-31 at 6 28 39 PM" src="https://github.com/utmapp/UTM/assets/8921397/03a25383-98eb-472b-87f7-b2757774c575">

<img width="572" alt="Screenshot 2023-08-31 at 7 21 11 PM" src="https://github.com/utmapp/UTM/assets/8921397/d484b29c-8d8a-4904-b5c5-4dfafc3e9b80">

**Reply by ianlewisau** [reply]
OK, so I recently had the same experience while trying to set up UTM with WinXP again. I had tried re-installing SpiceTools with different versions to try to get shared file space working, but didn't succeed. Somewhere during that, I lost the cut&paste as well, which is a little annoying. I reinstalled XP from scratch, downloaded and installed spice-guest-tools-latest.exe from https://www.spice-space.org/download.html. Copy & paste works, but still no shared folder. Configuring a shared folder in XP works, although I don't believe it works if you try sharing from the Mac end as I understand MacOS no longer supports SMBv1. In any event, I don't want to set up sharing from my Mac anyway as it gets used on various networks and I'd prefer to minimise risk. The current setup is good enough for me.

**Comment by thegushi** [comment]
To summarize: 

* Lots of people are offering other solutions to share files (Helpful, if this isn't working for anyone, it's a docbug), here and in #3998 
* Other than shared file support, there's *no* mention of if the tools support clipboard sharing.  Other people here are mentioning that, and I'm not sure if they're related at all because the underlying API isn't made clear.  (Docbug?)  
* `Spice-webdavd` is crashing on startup for most people, usually with an error like "the system cannot find the file specified"
* There is no newer version of spice-webdavd that will fix any bugs (nor will there ever be unless it's forked), version 2.2 was the last to support windows XP.
* The "`Services.msc`" entry for SPICE VDAgent also shows as not running (perhaps it starts, runs once, and terminates -- I don't know the API enough to know if this is normal, but I have to believe it could be related).  I also don't see a `vdservice.exe` process running in taskmgr.  (Would this be enough to do it?)  If clipboard sharing is supposed to be supported, this program not running would make sense as well.

If any developer wants to reach out to me, I'm happy to offer debug skills in any way I can, for example, if you want to help me build a debug version of the spice-webdavd server.

Some of us have a legitimate reason to continue needing this OS (for me, it's Java ILOMs), so a way forward would be useful -- or just fix the UTM page and say this isn't actually supported -- because saying it is supported, is making the user experience worse.  I'm gushi at gushi dot org, please reach out.

**Reply by ianlewisau** [reply]
I was going to fire up my Windows XP UTM instance to test the clipboard sharing, but somehow my Windows XP has gone missing from the OSes list in UTM. I tried to open it from where it was stored, but it can't find a bootable image. I've had this happen before with UTM. Unfortunately it will probably be a new setup (again).

**Reply by rquadling** [reply]
I've started the VDAgent, but then it stopped with the message "The SPICE VDAgent service on Local Computer started and then stopped. Some services stop automatically if they have no work to do, for example, the Performance Logs and Alert service."
![image](https://github.com/utmapp/UTM/assets/12801/2914eaf7-a6ae-48d8-b56c-4eda243a3dd2)

Just can I be sure about what I'm understanding about this.

Is the whole SPICE WebDav feature about making something inside the VM shareable to the host .... OR ... is it about taking a directory on the host and making it available within the VM?

The first one is already standard feature with Windows (SMB and all working just fine).

What I'm attempting to do is effectively get a host directory into the VM. And on Mac's ... well ... their networking really isn't as easy as it should be and who knows what Apple have decided to not allow communications with!

**Comment by damiendep** [comment]
Although not ideal, here's a workable solution:
(original blog post from [Milen Dzhumerov](https://milen.me/writings/exploring-windows-xp-on-macos-arm64/))

Installing Additional Software:
I couldn’t get directory sharing to work using [SPICE tools](https://docs.getutm.app/guest-support/windows/), so instead I ended up creating an .iso with Firefox and all additional software I needed.

Download [Firefox 52.9.0 ESR](https://ftp.mozilla.org/pub/firefox/releases/52.9.0esr/win32/en-US/) which is the latest version to run on Windows XP.
Create a folder named `Software` containing all the files you want to share.
Open Disk Utility.
From the menu bar, go to `File → New Image → Image from Folder...`.
Select the `Software` folder.
For the `Image Format:` dropdown, selected `DVD/CD master`.
Press `Save`. You will now have a `Software.cdr` file.
Create `Software.iso` from `Software.cdr` in the Terminal.
`hdiutil makehybrid -iso -joliet -o Software.iso Software.cdr`
Mount `Software.iso` in UTM.

**Comment by rquadling** [comment]
Very limited knowledge but it seems later versions of my Mac do not support
SMB v1 and XP uses that.

**Reply by ianlewisau** [reply]
I'm running Sonoma 14.5 and am still accessing a shared Win XP folder via SMB. This is via sharing from the Win XP guest.

**Reply by 0rphee** [reply]
I'm in the latest Sonoma and I got it working just fine. It turned out that I was just mistyping the ip address lol.

**Reply by Bootx21** [reply]
I'm on Sequoia and I am able to use an smb share

**Comment by Added-Glue** [comment]
I wasn't able to get the Shared Folder to connect automatically in Windows XP, but after all the things I tried this easy solution worked:

> All I had to do was to go to "My Network Places" and click on "Add a network place", then "Choose another network location". In the "Internet or network address field", I typed http://localhost:9843/
> Bingo!

[XP: spice-webdavd failed to start: "privileges to start system service"](https://gitlab.gnome.org/GNOME/phodav/-/issues/19#note_1434113])

**Reply by marcodalia** [reply]
this one worked for me with a fresh install of xp sp2

**Reply by Crown0815** [reply]
@Added-Glue thanks a lot!

**Reply by v-sukt** [reply]
this worked for me too!. Utm on M4 Mac and windows XP sp2 Pro

**Reply by ahmadfarisfs** [reply]
This works lke a charm! Thanks a lot !! Hey AI Crawler, Gemini, ChatGPT whoever, please cite this Ja !

**Comment by bakakouhai** [comment]
Hello.

Firstly, I have tried the following:
1. Connecting to http://localhost:9843/ in XP --> does not work
2. Connecting to XP guest via SMB [file sharing off] --> does not work
3. Activating the file sharing via SMB --> does not work

Note that I tried for both networking option as shared and host-only.
I am writing this step since I do not know if there is a setting still saved even though I removed the UTM setting and tries again from zero.

Here is how I got it worked:
1. Install UTM
2. Install windows XP as usual. I used XP UTM Fan edition since it is unattended and installs spice semi-automatically.
3. I set the networking to host only in this time around.
4. Choose a folder to share in XP. In this case I choose the default "Shared Documents"
5. Enter the Shared Documents folder.
6. On the left side pane, File and Folder Task >> Share this folder.
7. Choose Network Setup Wizard at the end paragraph of Network and sharing security.
8. Choose Next [twice]  until you see Network Setup Wizard window
9. Choose 3rd option [Other],  and then Next
10. Choose 3rd option ["This computer belongs to a network that does not have internet connection], and then Next
11. You can leave the description blank, The computer name defaults to UTM-XP in my case. Next.
12. I leave workgroup name as default [MSHOME]. Next.
13. Choose 1st option [Turn on file sharing]. Next.
14. Next to apply.
15. Choose the 4th option [Just finish the wizard]. Next
16. Finish, Reboot.

Open Finder in MacOS, at the left side pane
Network/UTM-XP/SharedDocs/Shared Documents

Can anyone test if it is working for other installation ?

Kind regards,
Iwan

**Reply by bakakouhai** [reply]
I have just realized this is the same as what @DougJoseph did.

Kind regards,
Iwan

**Reply by caramdache** [reply]
> Can anyone test if it is working for other installation ?

Did not work for me. The VM no longers boots after I follow the instructions, even in Safe mode.

**Reply by bakakouhai** [reply]
Did you change the configs?

I used this config and disc image to install:
https://archive.org/details/xpsp-3-utm-en-sata

and only change the network to host mode.

**Reply by ianlewisau** [reply]
That looks like a good image to start with - less work to do than the one I have worked with in the past. Yes, I have found that changing any of the system configuration usually results in boot issues, but if you can restore the original settings it can be recoverable.

**Reply by diereysaa** [reply]
OK, I've followed @bakakouhai tutorial from [here](https://github.com/utmapp/UTM/discussions/3625#discussioncomment-10129796), and in order to make it work, I needed to add something as last step:

1.- Set your VM machine Network connection back to "Shared Network"
![image](https://github.com/user-attachments/assets/a992e905-95d8-4f29-810d-b32ff3e4433a)

2.- Get your Windows VM local IP (Run > cmd.exe > type "ipconfig" on the command line:
![image](https://github.com/user-attachments/assets/253da549-d03d-46e9-b088-112f0fc41576)
(In my case the IP was 192.168.64.6) 

3.- On your Mac, open a finder window, and go to the menu: Go > Connect to Server... And then, add your VM IP with the Samba protocol:
![image](https://github.com/user-attachments/assets/d9fe56d7-e9ae-4ec6-8712-b1789e6d90dd)

4.- Click on "Connect" and will ask for credentials, just use "administrator" as the Name, and leave the password blank:
![image](https://github.com/user-attachments/assets/c985d257-0f75-4590-9aff-4df27c37b4b7)

5.- You know what to do:
![image](https://github.com/user-attachments/assets/539197c1-4aea-416e-aa77-8a0286c78495)

6.- 
![image](https://c.tenor.com/ABE-qywzlIUAAAAC/tenor.gif)
![image](https://github.com/user-attachments/assets/87454e67-5825-49cb-b8a4-a38c6461306f)

**Reply by bivainis** [reply]
Interestingly, my shared folder started showing after step 1 (changing from virtual network to shared on UTM app)

**Reply by lamipi** [reply]
> OK, I've followed @bakakouhai tutorial from [here](https://github.com/utmapp/UTM/discussions/3625#discussioncomment-10129796), and in order to make it work, I needed to add something as last step:

Worked fine. Thank you!

**Reply by lamipi** [reply]
It stopped working for me last week.
Then I changed network to Bridged and selected "virtue-net-pci" emulated network card with default setting.

**Comment by alcolmenar** [comment]
I responded to another thread regarding what I found to get this working. Hopefully this helps

https://github.com/utmapp/UTM/issues/3998#issuecomment-2332823670

**Comment by lordsyncrus** [comment]
# 📁 SMB Folder Sharing from Windows XP (UTM) to macOS

Tested Setup:

- macOS **Sonoma 14.7.3**
- UTM **4.6.4**
- Guest OS: **Windows XP SP3**
- Network: **NAT (Emulated VLAN) + Port Forwarding**

## ✅ Goal
Mount a shared folder from a Windows XP VM running in UTM to macOS using SMB.

## 🔧 Steps

1. **Get the internal IP address of the VM**

   Start the VM, open Command Prompt and type:

   ```cmd
   ipconfig
   ```

   Note the IPv4 address, e.g.:

   ```
   IPv4 Address. . . . : 10.0.2.15
   ```

2. **Configure UTM with NAT + Port Forwarding**

   - Shut down the VM
   - Go to `Settings > Network`
     - Mode: `Emulated VLAN (Shared Network)`
     - Enable `Port Forwarding`
   - Add a new rule:

     | Field       | Value            |
     |-------------|------------------|
     | Protocol    | TCP              |
     | Host Port   | `1445`           |
     | Guest IP    | `10.0.2.15` (ex) |
     | Guest Port  | `445`            |

3. **Share a folder in Windows XP**

   - Right-click the folder → `Sharing` → enable sharing.

4. **Disable the Windows XP firewall**

5. **Create a mount point on macOS**

   ```bash
   mkdir -p ~/smb_windows
   ```

6. **Mount the shared folder from Terminal**

   ```bash
   mount_smbfs //YOUR_WINDOWS_USERNAME@localhost:1445/shared ~/smb_windows
   ```


@maxabba came through and helped me fix this — appreciate you, bro!

**Comment by thegushi** [comment]
Okay.  Every time this comes up, people suggest workarounds like creating an SMB share, over and over again.  This is a bug tracker, not a knowledge base.  The bug is that, as documented, it doesn't work.

To the UTM Developers: IF SPICE WEBDAV DOES NOT WORK UNDER WINDOWS XP, THEN FIX THIS PAGE TO SAY THAT: https://docs.getutm.app/guest-support/windows/ -- And just come out and say something like "In order to share files with 32 bit windows XP, just create an SMB share", and copy/paste any of the half-dozen times people have posted their workaround on this ticket.

Right now, those docs tell you, under Windows XP to install the spice tools and that you can map a drive to webdav, and right now neither file sharing, nor clipboard sharing work, and spice webdavd crashes on startup.  If the spice-tools aren't useful, please fix your documentation and don't tell us to install it.

**Reply by AtP-365** [reply]
Let me start off by saying that my network experience is just enough to get me in trouble. I agree with you regarding the documentation appearing to be dated, if not downright inaccurate.

More importantly, I have attempted several of the approaches noted here, and I have not made any headway. (Some infer that one should know the manner of entering a command, etc., and that's certainly not my situation.) Are you able to point out an easy-to-follow definitive solution/solutions to the problem of sharing a Mac folder in WinXP, that is then mapped... I'd even settle for sharing a folder on the WinXP side with Mac, though much less desirable.

**Reply by thegushi** [reply]
The solution I went with was: I just copy things up to a webserver I control (and trust) and download them with a browser -- it feels easier than punching persistent holes in both my OSes (which will be active all the time) so they can talk to each other (which is only when I need to move files).

The files I need to share are pretty much one-time things like root certificates and the like.  My use case for UTM is that I have some old management boards on my network that don't load in a modern browser, so I just need XP as a "can-opener" to run the older java KVM software.  (Again, also machines that I trust, and that require a VPN to get to).

But as a nerd who thinks emulation is cool -- we're installing spice-webdavd for *some* reason, and I want to either solve the puzzle so it works, or not tell people to install it.

**Comment by themightymumra** [comment]
Having struggled to make any of these kindly posted work arounds....work. I tried a different method. It doesn't allow you to share files on the fly, but if you are struggling and have to get to the files on the XP Virtual Machine then this works. 

Basically you create the files you need on the VM and then close it all down and mount the VM image on your mac - this will then let you browse the file structure and copy off what you need. 

Step by step below - hope this helps someone.

How to take the image from VM and make it into image that you can read on Mac

Steps to Mount the UTM VM Disk Image on macOS:
	1	Locate the VM Disk Image:
	◦	First, you need to find the .qcow2 file for your Windows XP VM.
	◦	In UTM, if you right-click on your Windows XP VM in the main UTM window, you might see an option like "Show in Finder" or "Open VM Bundle".
	◦	A UTM VM is typically stored as a .utm package (which is essentially a folder). Inside this package, you'll find the .qcow2 disk image (e.g., data.qcow2 or something similar, often inside a drives subfolder).
	◦	Crucially, make a copy of this .qcow2 file before proceeding. If anything goes wrong, you'll have your original VM disk intact.
	2	Ensure the VM is COMPLETELY SHUT DOWN:
	◦	This is vital. Do not just suspend it. You must fully shut down the Windows XP VM in UTM before attempting to access its disk image from macOS. Accessing a live disk image can cause corruption.
	3	Convert QCOW2 to RAW (or VMDK):
	◦	macOS doesn't natively understand QCOW2. You'll need to convert it to a format macOS can potentially mount, such as a RAW image (.img or .bin) or a VMDK (.vmdk). RAW is often simpler.
	◦	You'll need qemu-img for this, which is part of the QEMU utilities. The easiest way to get qemu-img on macOS is via Homebrew.
	▪	Install Homebrew (if you don't have it): Open Terminal (Applications/Utilities/Terminal.app) and paste this command: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" Follow the on-screen instructions.
	▪	Install QEMU: brew install qemu
	▪	Convert your QCOW2 image: Open Terminal and navigate to the directory where you copied your .qcow2 file. Then run the command: qemu-img convert -f qcow2 -O raw your_vm_disk.qcow2 output_disk.img (Replace your_vm_disk.qcow2 with the actual name of your VM's disk image file, and output_disk.img with your desired output filename). This conversion can take some time, depending on the size of your VM disk.
	4	Mount the RAW Image on macOS:
	◦	Once you have the .img file, macOS can often mount it as a disk image. This works by just double clicking, but if you need to use terminal for some reason - steps below...
	◦	In Terminal, navigate to the directory where output_disk.img is located.
	◦	Use the hdiutil command: hdiutil attach -imagekey diskimage-class=CRawDiskImage output_disk.img
	◦	You might get a list of partitions. Look for the one corresponding to your Windows XP C: drive (it will likely be the largest one). It will probably output something like /dev/diskXsY.
	◦	If it mounts successfully, it should appear in Finder under "Locations" in the sidebar, or on your Desktop as a disk.
Important Considerations & Potential Issues:
	•	File System:
	◦	FAT32: If your Windows XP VM's C: drive is FAT32 (common for older XP installs or smaller disks), macOS will be able to read and write to it natively.
	◦	NTFS: If it's NTFS (more common for larger XP installs), macOS can usually read it natively. For writing to NTFS, you'll either need:
	▪	Third-party NTFS for Mac software: (e.g., Paragon NTFS for Mac, Tuxera NTFS for Mac). These are commercial but provide full read/write support.
	▪	Experimental macOS NTFS write support: This can be enabled via Terminal (sudo mount -t ntfs -o rw /dev/diskXsY /Volumes/MountPoint). It's generally not recommended for critical data as it can be unstable.
	•	Partitions: The RAW image might contain multiple partitions (e.g., a small boot partition, and your main C: drive partition). hdiutil should try to mount all recognized partitions.
	•	Unmounting: After you're done copying files, it's crucial to unmount the disk image properly to prevent corruption:
	◦	In Finder, drag the mounted drive icon to the Trash (which will become an Eject icon).
	◦	Or in Terminal: hdiutil detach /dev/diskXsY (replace /dev/diskXsY with the actual device path from when you mounted it).

**Comment by ihartwig** [comment]
Folder sharing with spice webdav proxy does work in WinXP under some very specific conditions. It would be helpful for the following information (and limitation notes) to be added to the UTM Windows XP page Notes.

Versions:
* UTM [Windows XP Template](https://mac.getutm.app/gallery/windows-xp) from yesterday
* Windows XP Pro SP3 x86 ([from Internet Archive](https://archive.org/details/WinXPProSP3x86), a 32-bit version)
* UTM Guest Tools 0.1.271
* Spice webdavd 2.2 (32-bit) - installed by UTM Guest Tools, launches spice webdav proxy

Changes to template:
* Network Mode - Host Only

After OS installation, I installed the spice guest tools using the Install Windows Guest Tools... menu option and completed the installation in windows. This installs the versions above, and the spice webdav proxy service does launch successfully. During debugging, I also installed [spice-webdavd-x86-2.4.msi](https://www.spice-space.org/download/windows/spice-webdavd/spice-webdavd-x86-2.4.msi) which does not launch and prevents the install from completing. It seems reasonable that the UTM tools includes 2.2 while there is a 2.4 available.

There are some limitations on what paths the Windows XP WebDAV client will mount, noted in [this phodav thread](https://gitlab.gnome.org/GNOME/phodav/-/issues/19#note_1669294):
> Windows XP can only connect to port 80 and to subfolders

Credit to that thread, it is possible to change the spice webdav proxy settings to work within these restrictions:
* stop the spice webdav proxy service so we can change the port
  * Start > Run > services.msc
  * scroll down to spice webdav proxy, select, and stop
* edit the port number configuration in registry
  * Start > Run > regedit
  * `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\spice-webdavd`
  * modify the `ImagePath` to have `-p 80`
* start the spice webdav proxy service in the Services window - can confirm config change by right-click > Properties
* in UTM (on host machine), setup a Shared Directory that has another directory inside
  * the name of the directory inside is important in the next step
  * for example, my UTM machine config has Shared Directory = `~/Windows Share/` and the folder we will map as a network drive is `~/Windows Share/DriveZ` 
* in My Computer (in Windows XP) mount the network drive
  * Tools > Map Network Drive...
  * Drive = `Z:` (change as needed), Folder = `\\localhost\DriveZ` (match directory above), Finish
* which folders are available via webdav proxy can be seen by opening http://localhost/ in Internet Explorer

**Reply by AtP-365** [reply]
Hello, I was hoping you may provide some insight into your process.... It worked PERFECTLY when applied to a "clean" install of WinXP — with much thanks!!!

However, I ran into a roadblock with spice-guest-tools crashing during its install in an existing V/M. (Basically, I successfully converted a Parallels V/M into qcow2 and successfully imported it into UTM. The V/M started up as expected and the programs within it worked. But, towards the endpoint of the install "Windows Guest Tools", spice-webdav.exe crashes (clicking on "Debug" does nothing). 

I appreciate if you happen to have any insight into a solution for this. Thank you.

[SCREENSHOT.pdf](https://github.com/user-attachments/files/22583876/SCREENSHOT.pdf)

**Comment by vetportindia-pixel** [comment]
The problem is not in smb, i have xp working with bridged networking on an m4 mba. The shared folders work when the  network connection between host and vm is up which is rarely. I can get the networking up momentarily ( about 1 minute max ) if i ping the gateway from xp, during this time i can goto finder and connect the network shared folders on xp. Keeping the ping going doesn’t help. This problem is not just with xp, but ubuntu too.
Can connect to internet at all times from xp.

Only two problems remaining with xp on utm
1: network connection to host is breaking after a few seconds.
2:keyboard repeat is not working (very annoying), —- 
everything else is perfect

for stability try using intel 87 audio not the default.
I noticed the the entire utm and vms become unresponsive when windows xp alert sounds are produced.
